src/features/embalm/stepContent/hooks/useLoadArchaeologists.ts

```diff
@@ -1,48 +1,69 @@
-import { useAsyncEffect } from 'hooks/useAsyncEffect';
-import { generateMockArchaeoloigsts } from 'lib/mocks/mockArchaeologists';
-import { LibP2pContext } from 'lib/network/P2PNodeProvider';
-import { useContext, useEffect } from 'react';
+import { ViewStateFacet } from 'lib/abi/ViewStateFacet';
+import { useNetworkConfig } from 'lib/config';
+import { useEffect } from 'react';
 import { startLoad, stopLoad } from 'store/app/actions';
-import { setArchaeologists, setSelectedArchaeologists } from 'store/embalm/actions';
-import { useDispatch } from 'store/index';
-import { Archaeologist } from 'types/index';
+import { setArchaeologists } from 'store/embalm/actions';
+import { useDispatch, useSelector } from 'store/index';
+import { ArchaeologistProfile } from 'types';
+import { readContract } from 'wagmi/actions';
 
+/**
+ * Loads archaeologist profiles from the sarcophagus contract
+ */
 export function useLoadArchaeologists() {
   const dispatch = useDispatch();
-
-  const browserNode = useContext(LibP2pContext);
+  const networkConfig = useNetworkConfig();
+  const { archaeologists } = useSelector(s => s.embalmState);
 
   useEffect(() => {
     (async () => {
       try {
+        // Only load the archaeologists once when the component mounts. The only reason the
+        // archaeologists would need to be loaded from the contract again is when a new
+        // archaeologist registers.
+        if (archaeologists.length > 0) return;
         dispatch(startLoad());
-        const loadedArchaeologists: Archaeologist[] = generateMockArchaeoloigsts();
 
-        // TODO: Temporarily remove this and load mock archaeologists
-        // if (browserNode === undefined) {
-        //   console.error('browser node is undefined');
-        //   return;
-        // }
+        let profiles: Record<string, ArchaeologistProfile> = {};
```

Comment:
This was copied over from Kelvin's work and has probably lost it's purpose since then. It's a valid question though. However, this will be changed to an array in my current PR when we load the profiles directly from the contract.

Reply:
IIRC this was to make it easier to reference this 'list' simply by using a key instead of a `find`. But yeah it's probably lost its purpose since Kyle's updates. I don't even remember the original reason for said referencing lol.

Reply:
If we are looking up a profile by its address (or whatever `string` is here), then this should remain as a `Record` or maybe a `Map`.